Sources/MapboxCoreNavigation/RoutingProvider.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+import MapboxDirections
+
+/**
+ Protocol which defines a type which can be used for fetching or refreshing routes.
+ 
+ SDK provides conformance to this protocol for `Directions` and `MapboxRoutingProvider`.
+ */
+public protocol RoutingProvider {
+    
+    /**
+     Routing caluclation method.
+     
+     - parameter options: A `RouteOptions` object specifying the requirements for the resulting routes.
+     - parameter completionHandler: The closure (block) to call with the resulting routes. This closure is executed on the application’s main thread.
+     - returns: Related request. If, while waiting for the completion handler to execute, you no longer want the resulting routes, cancel corresponding task using this handle.
+     */
+    @discardableResult func calculateRoutes(options: RouteOptions,
```

Comment:
Looks like `MapboxRoutingProvider` also has documentation comments. Does it make sense to merge them to the base protocol? Xcode automatically provides documentation from from the protocol if you implement the protocol.

Reply:
Documentation differs for `RoutingProvider` and `MapboxRoutingProvider`. Protocol description is more abstract, defining the purpose of each method, while the latter one provides more insights on how the method does it's job.